pkg/ss/handler.go

```diff
@@ -34,7 +34,7 @@ func (mh *MsgHandler) Handle(req Msg) (interface{}, error) {
 	if err := mh.OnMsg(req); err != nil {
 		// TODO find a better place to close this
 		close(req.ResponseChannel())
-		return "", status.Errorf(codes.Unavailable, err.Error())
+		return "", status.Errorf(codes.Internal, err.Error())
```

Comment:
Return the same error that you got. No need to wrap it. For starters you don't know if it's internal or unavailable

Reply:
if we don't specify, they just show up as unknown code in response. but i understand the challenge in categorising the errors

Reply:
You should see a proper code unless upstream is not throwing the correct error